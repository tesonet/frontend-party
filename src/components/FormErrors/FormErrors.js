import React from "react";
import PropTypes from "prop-types";

const FormErrors = ({ children }) => {
  switch (typeof children) {
    case "string":
      return (
        <div className="FormErrors">
          <p>{children}</p>
        </div>
      );
    case "object":
      return (
        <ul className="FormErrors">
          {Object.keys(children).map(i => (
            <li key={i}>
              <p>{children[i]}</p>
            </li>
          ))}
        </ul>
      );
    default:
      break;
  }
};

FormErrors.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.object])
};

export default FormErrors;
