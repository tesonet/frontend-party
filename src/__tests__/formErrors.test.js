import React from "react";
import renderer from "react-test-renderer";

//Component.
import FormErrors from "../components/FormErrors/FormErrors";

it("renders correctly string error", () => {
  const tree = renderer
    .create(<FormErrors>"This is string error"</FormErrors>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly array of strings", () => {
  const errorMock = {
    1: "Username is required",
    2: "Password is required"
  };

  const tree = renderer.create(<FormErrors>{errorMock}</FormErrors>).toJSON();
  expect(tree).toMatchSnapshot();
});
