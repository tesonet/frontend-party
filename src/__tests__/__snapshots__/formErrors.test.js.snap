// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly array of strings 1`] = `
<ul
  className="FormErrors"
>
  <li>
    <p>
      Username is required
    </p>
  </li>
  <li>
    <p>
      Password is required
    </p>
  </li>
</ul>
`;

exports[`renders correctly string error 1`] = `
<div
  className="FormErrors"
>
  <p>
    "This is string error"
  </p>
</div>
`;
